### ISSUE TITLE
Run checksums against models?

### ISSUE THREAD

**Issue by bmschmidt** [nomic-ai/pyllamacpp#2]
Our distributed models include checksums. Rather than urge users to run them--which they won't?--a safety conscious approach might be to hardcode some checksums on models and on compiled files into the code?

**Comment by abdeladim-s** [comment]
but when we should include the checksum, on every Model instanciation ?

**Comment by bmschmidt** [comment]
Not workable because it takes too long... I was thinking, cache a file somewhere indicating it's been checked? But TBH this is probably low-priority enough to be closable--it stems from our concerns about distributing executable files like we've been doing the last week, but the downloaded models don't present any substantial risk that I can see? And any compiled files in the wheels are pretty much the same as any other python code.

**Comment by abdeladim-s** [comment]
Exactly, the checksum should be included if we implemented the auto-downloading feature you mentioned in the previous issue. 

I saw your previous approach in the `nomic` client, distributing executable is not the right thing to do, besides it is difficult to maintain, it may flag antiviruses if they are not signed. This will trigger many concerns for the end users. 

As you said, The wheels are just Python code, so there are no substantial risk. 
Feel free to close it, or link it to the other issue.
